packages/sources/tp/src/endpoint/price.ts

```diff
@@ -9,53 +9,61 @@ import { WebsocketTransportGenerics } from '@chainlink/external-adapter-framewor
 import { InputParameters } from '@chainlink/external-adapter-framework/validation'
 import { generateTransport } from '../transport/price'
 
+export type QueryParams = {
+  streamName: 'TP' | 'IC'
+}
+
 export type BaseEndpointTypes = WebsocketTransportGenerics & {
   Parameters: typeof inputParameters.definition
   Response: SingleNumberResultResponse
   Settings: typeof config.settings
 }
 
-export type GeneratePriceOptions = {
-  sourceName: 'tpSource' | 'icapSource'
-  streamName: 'TP' | 'IC'
-  sourceOptions?: string[]
-}
-
-export const generateInputParams = (
-  generatePriceOptions: GeneratePriceOptions,
-): InputParameters<PriceEndpointInputParametersDefinition> =>
+export const generateInputParams = (): InputParameters<PriceEndpointInputParametersDefinition> =>
   new InputParameters(
     {
       ...priceEndpointInputParametersDefinition,
-      [generatePriceOptions.sourceName]: {
-        description: `Source of price data for this price pair on the ${generatePriceOptions.streamName} stream`,
+      streamName: {
+        aliases: ['source'],
+        description: 'TP or ICAP',
```

Comment:
For avoiding any confusion whether ICAP == IC

```suggestion
        description: "TP ('TP') or ICAP ('IC')",
```